pyiceberg/io/__init__.py

```diff
@@ -46,6 +48,10 @@
 
 logger = logging.getLogger(__name__)
 
+AWS_REGION = "client.region"
```

Comment:
I chose `client.` for unified configurations because in java:
https://github.com/apache/iceberg/blob/c68abfc9fd3956077b43aba20441f089bb8b93d6/aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java#L67
`client.region` is used for configurations that affect all the clients.

Reply:
Nice, that's a good find 👍

Reply:
would be good to highlight this in the PR description!